src/Datasets/PreprocessorRepository.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Pest\Datasets;
+
+use InvalidArgumentException;
+use Pest\Contracts\Dataset\Preprocessor;
+
+/**
+ * @internal
+ */
+final class PreprocessorRepository
```

Comment:
My eventual idea is to register this in the Pest container, then inject it into the `TestCaseDataSet`

Reply:
I think this will be the greatest feature of this PR